pyalex/api.py

```diff
@@ -169,6 +170,30 @@ class OpenAlexEntity(dict):
     pass
 
 
+class OpenAlexResponseList(list):
+    """A list of OpenAlexEntity objects with metadata.
+
+    Attributes:
+        meta: a dictionary with metadata about the results
+        resource_class: the class to use for each entity in the results
+
+    Arguments:
+        results: a list of OpenAlexEntity objects
+        meta: a dictionary with metadata about the results
+        resource_class: the class to use for each entity in the results
+
+    Returns:
+        a OpenAlexResponseList object
+    """
+
+    def __init__(self, results, meta=None, resource_class=None):
```

Comment:
You might as well use `OpenAlexEntity` as the default value for `resource_class`.

Reply:
Makes sense. Always not completely sure if this is a good practice (mutable vs immutable).

Reply:
The class is immutable, but the instances are not. So you should not have `OpenAlexEntity()` as default, but `OpenAlexEntity` is fine.

Reply:
Thanks. Great. Will update